pkg/host-disk/host-disk.go

```diff
@@ -170,13 +172,30 @@ func dirBytesAvailable(path string, reserve uint64) (uint64, error) {
 	return stat.Bavail*uint64(stat.Bsize) - reserve, nil
 }
 
-func createSparseRaw(fullPath string, size int64) (err error) {
+func createSparseRaw(diskdir, fullPath string, size int64) (err error) {
 	offset := size - 1
-	f, err := os.Create(fullPath)
+	safeDiskdir, err := safepath.NewPathNoFollow(diskdir)
+	if err != nil {
+		return err
+	}
+
+	err = safepath.TouchAtNoFollow(safeDiskdir, filepath.Base(fullPath), 0666)
+	if err != nil {
+		return err
+	}
+
+	sFile, err := safepath.NewFileNoFollow(fullPath)
+	if err != nil {
+		return err
+	}
+	defer util.CloseIOAndCheckErr(sFile, &err)
+
+	f, err := os.OpenFile(sFile.SafePath(), os.O_WRONLY, 0666)
```

Comment:
Why not `safepath.OpenAtNoFollow`?

Reply:
I didn't find it better because anyway I need safepath as input arg... , `OpenAtNoFollow` would again do a lot of `openat`...

Reply:
Ack. Test failures seem relevant: `preparing host-disks failed: failed opening var for path root: /, relative: /proc/31431/root/var/run/kubevirt-private/vmi-disks/disk0/disk.img: not a directory`